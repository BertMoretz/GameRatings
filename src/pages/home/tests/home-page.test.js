import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import axios from 'axios';
import moxios from 'moxios';

import HomePage from '../home-page'

describe('HomePage', () => {
  it('test render empty component', () => {
    const initialState = {

    }

    const container = renderer
      .create(
          <Router>
            <Route component={HomePage}/>
          </Router>,
      )

    expect(container.toJSON()).toMatchSnapshot()
  }),

  it('test render non empty component', () => {
    const initialState = {
      games: [{
        "id": 36950,
        "aggregated_rating": 61.0,
        "cover": {
          "id": 71030,
          "alpha_channel": false,
          "animated": false,
          "game": 36950,
          "height": 960,
          "image_id": "co1it2",
          "url": "//images.igdb.com/igdb/image/upload/t_thumb/co1it2.jpg",
          "width": 677
        },
        "genres": [
          {
            "id": 5,
            "created_at": 1297555200,
            "name": "Shooter",
            "slug": "shooter",
            "updated_at": 1323216000,
            "url": "https://www.igdb.com/genres/shooter"
          },
          {
            "id": 12,
            "created_at": 1297555200,
            "name": "Role-playing (RPG)",
            "slug": "role-playing-rpg",
            "updated_at": 1323216000,
            "url": "https://www.igdb.com/genres/role-playing-rpg"
          },
          {
            "id": 31,
            "created_at": 1323561600,
            "name": "Adventure",
            "slug": "adventure",
            "updated_at": 1323561600,
            "url": "https://www.igdb.com/genres/adventure"
          }
        ],
        "name": "Anthem",
        "platforms": [
          6,
          48,
          49
        ],
        "summary": "Anthem is a shared-world action RPG, where players can delve into a vast landscape teeming with amazing technology and forgotten treasures. This is a world where Freelancers are called upon to defeat savage beasts, ruthless marauders, and forces plotting to conquer humanity."
      }]
    }

    const container = renderer
      .create(
          <Router>
            <Route component={HomePage}/>
          </Router>,
      )

    expect(container.toJSON()).toMatchSnapshot()
  }),

  it('test api', () => {

      let onFulfilled = jest.fn()

      moxios.stubRequest('https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games', {
          status: 200,
          response: {
              results: [{ test: 'game' }]
          }
      })

      axios.get('https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games').then(onFulfilled)

      moxios.wait(function () {
        expect(onFulfilled).toBeCalled();
        expect(onFulfilled.mock.calls[0][0]).toMatchSnapshot();
        done()
      })
  })

  afterEach(() => {
      moxios.uninstall()
  })
})
