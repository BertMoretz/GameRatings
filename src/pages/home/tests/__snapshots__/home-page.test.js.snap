// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage test render empty component 1`] = `
<div
  className="loading"
>
   
  <div
    className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-6"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
   
</div>
`;

exports[`HomePage test render non empty component 1`] = `
<div
  className="cards"
>
  <ul
    cellSpacing={true}
    className="MuiGridList-root-11 gridList"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root-12"
      style={
        Object {
          "height": "auto",
          "padding": 2,
          "width": "100%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile-13"
      >
        <div
          className="MuiListSubheader-root-16 MuiListSubheader-sticky-21 MuiListSubheader-gutters-19"
        >
          Popular Now
        </div>
      </div>
    </li>
    <li
      className="MuiGridListTile-root-12 elem"
    >
      <div
        className="MuiGridListTile-tile-13"
      >
        <a
          className="coverlink"
          href="#"
          onClick={[Function]}
        >
          <div
            className="detailscover"
          >
            Details
          </div>
          <img
            alt="Anthem"
            className="elem"
            src="https://images.igdb.com/igdb/image/upload/t_cover_big/co1it2.jpg"
          />
          <div
            className="MuiGridListTileBar-root-22 MuiGridListTileBar-titlePositionBottom-23 MuiGridListTileBar-rootSubtitle-25 caption"
          >
            <div
              className="MuiGridListTileBar-titleWrap-26 MuiGridListTileBar-titleWrapActionPosRight-28"
            >
              <div
                className="MuiGridListTileBar-title-29"
              >
                Anthem
              </div>
              <div
                className="MuiGridListTileBar-subtitle-30"
              >
                <span>
                   
                  Shooter
                </span>
              </div>
            </div>
            <div
              className="MuiGridListTileBar-actionIcon-31"
            >
              <button
                className="MuiButtonBase-root-39 MuiIconButton-root-33"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-38"
                >
                  <div
                    className="score"
                  >
                    61%
                  </div>
                </span>
                <span
                  className="MuiTouchRipple-root-42"
                />
              </button>
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
